File: Count the characters - GFG/count-the-characters.java
// { Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;
class GfG
{
    public static void main (String[] args)
    {
        
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while(t-- > 0)
        {
            String s = sc.next ();
            int n = sc.nextInt();

    		System.out.println (new Sol().getCount (s, n)); 
        }
        
    }
}

// Contributed By: Pranay Bansal
// } Driver Code Ends


//User function Template for Java

class Sol
{
    int getCount (String s, int N)
    {
        // your code here
        Map<Character, Integer> map = new HashMap<>();
        map.put(s.charAt(0), 1);
        for(int i=1; i<s.length(); i++){
            if(s.charAt(i-1) != s.charAt(i)){
                map.put(s.charAt(i), map.getOrDefault(s.charAt(i),0)+1);
            }
        }
        
        int count = 0;
        for(char c : map.keySet()){
            if(map.get(c) == N)
            count++;
        }
        
        return count;
    }
}

File: Count the characters - GFG/count-the-characters.cpp
// { Driver Code Starts
#include<bits/stdc++.h>
using namespace std;

 // } Driver Code Ends

class Solution
{
  public:
        int getCount (string S, int N)
        {
            //code here.
            unordered_map<char,int>ans;
            // for(auto it:S){
            //     ans[it]++;
            // }
            ans[S[0]]++;
            for(int i=1; i<S.size(); i++){
                if(S[i] != S[i-1]){
                     ans[S[i]]++;
                }
            }
            int cnt=0;
            for(auto i: ans){
                // cout<<i<<" ";
                if(i.second == N){
                    cnt++;
                }
            }
            return cnt;
        }
};

// { Driver Code Starts.
int main()
{
    int t; cin >> t;
    while (t--)
    {
        string s; cin >> s;
        int n; cin >> n;
        
        Solution ob;
        cout <<ob.getCount (s, n) << endl;
    }
}

// Contributed By: Pranay Bansal
  // } Driver Code Ends